source/patterns/@aws-solutions-constructs/aws-dynamodb-stream-lambda/lib/index.ts

```diff
@@ -69,7 +69,7 @@ export interface DynamoDBStreamToLambdaProps {
 
 export class DynamoDBStreamToLambda extends Construct {
   public readonly lambdaFunction: lambda.Function;
-  public readonly dynamoTable: dynamodb.Table;
+  public readonly dynamoTable?: dynamodb.Table;
```

Comment:
I'd expose both this as an optional property as well as the ITable, which will always exist. Then we need to document well when the Table object would be available.

Reply:
Are you suggesting something like this:

```
  public readonly dynamoTable?: dynamodb.Table;
  public readonly dynamoTableInterface?: dynamodb.ITable;
```

where `dynamoTable` will be available if the user user provides `dynamoTableProps` or `existingTableObj` of type `Table` and `dynamoTableInterface` will be available if the user provides `existingTableObj` of type `ITable` ?

Reply:
I was thinking:
```
  public readonly dynamoTable?: dynamodb.Table;
  public readonly dynamoTableInterface: dynamodb.ITable;
```

(making as much info as possible available). But I hadn't thought about names, that's unfortunate. 

As I recall, if the client supplies existingTableObject the code currently returns ITable regardless of what was passed to the ITable parameter. This is actually pretty sticky - why don't you set up a meeting tomorrow morning for you, Ryan and me to toss around ideas. The cleanest solution would be to go ITable for everything about this construct - the only drawback is that if someone needs the table somewhere else they can't use it. The workaround would be to create the table somewhere else (either independently or as part of another construct).

Reply:
Yep, cannot go with `ITable` for everything then the user won't be able to make mutating calls on `dynamoTable` property e.g. `dynamoTable.addGlobalSecondaryIndex()`, also it will break the existing customer's code..